src/enrichment/models/cider.rs

```diff
@@ -150,4 +128,270 @@ impl CiderImagesModel {
+    pub fn photometry_inputs(
+        &self,
+        photometry: Vec<PhotometryMag>,
+    ) -> Result<
+        (
+            ArrayBase<OwnedRepr<f32>, Dim<[usize; 3]>>,
+            ArrayBase<OwnedRepr<bool>, Dim<[usize; 2]>>,
+        ),
+        ModelError,
+    > {
+        let mut sorted_photometry = Vec::from(photometry);
+        sorted_photometry.sort_by(|a, b| a.time.partial_cmp(&b.time).unwrap());
+        sorted_photometry.dedup_by(|a, b| a.time == b.time && a.band == b.band);
+        let mut result = Vec::new();
+
+        let mut bands: std::collections::HashMap<String, Vec<PhotometryMag>> =
+            std::collections::HashMap::new();
+        for point in sorted_photometry {
+            bands
+                .entry(point.band.clone())
+                .or_insert_with(Vec::new)
+                .push(point);
+        }
+        // Process each band separately
+        for (_band, mut points) in bands {
+            // Sort by time
+            points.sort_by(|a, b| a.time.partial_cmp(&b.time).unwrap());
+            let mut i = 0;
+            while i < points.len() {
+                let mut group = vec![&points[i]];
+                let mut j = i + 1;
+
+                // Collect all points within time_threshold of the first point in group
+                while j < points.len() && (points[j].time - points[i].time) <= 0.5 {
+                    group.push(&points[j]);
+                    j += 1;
+                }
+
+                // Combine the group
+                let combined = Self::combine_group(&group);
+                result.push(combined);
+
+                i = j;
+            }
+        }
+        // Sort final result by time
+        result.sort_by(|a, b| a.time.partial_cmp(&b.time).unwrap());
+
+        let initial_t = result[0].time;
+        let mut dt = Vec::new();
+        let mut delta_t = Vec::new();
+        let mut temp = result[0].time;
+        let mut log_fluxes = Vec::new();
+        let mut sigma_fluxes = Vec::new();
+        let mut tail1: Vec<f32> = Vec::new();
+        let mut tail2: Vec<f32> = Vec::new();
+        let mut tail3: Vec<f32> = Vec::new();
+
+        for point in &result {
+            let diff_delta_t = point.time - initial_t;
+            let diff_dt = point.time - temp;
+            temp = point.time;
+            dt.push(diff_dt as f32);
+            delta_t.push(diff_delta_t as f32);
+            log_fluxes.push(point.mag.max(1e-6).log10() as f32);
+            sigma_fluxes.push(point.mag_err / (point.mag * 2.302585) as f32);
+            if point.band == "g" {
+                tail1.push(1.0);
+                tail2.push(0.0);
+                tail3.push(0.0);
+            } else if point.band == "r" {
+                tail1.push(0.0);
+                tail2.push(1.0);
+                tail3.push(0.0);
+            } else {
+                tail1.push(0.0);
+                tail2.push(0.0);
+                tail3.push(1.0);
+            }
```

Comment:
maybe it would be a lot more efficient to create tail1, tail2, tail3 as arrays of the correct size to begin with (so maybe they don't have to be vecs anymore, just regular arrays) filled with 0s, and add 1s where needed? That way you just write to one already pre-allocated array at a time, instead of allocating more and more memory to 3 arrays